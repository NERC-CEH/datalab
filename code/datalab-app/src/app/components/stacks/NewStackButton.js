import React from 'react';
import PropTypes from 'prop-types';
import Card from '@material-ui/core/Card';
import Fab from '@material-ui/core/Fab';
import Icon from '@material-ui/core/Icon';
import Tooltip from '@material-ui/core/Tooltip';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
  card: {
    height: '100%',
    backgroundColor: 'transparent',
  },
  buttonContainer: {
    width: '100%',
    height: '100%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'left',
  },
  button: {
    margin: 20,
  },
  icon: {
    fontSize: 40,
  },
});

const NewStackButton = ({ classes, onClick, typeName }) =>
  <Card className={classes.card} elevation={0}>
    <div className={classes.buttonContainer}>
      <Tooltip title={`Create ${typeName}`}>
        <Fab className={classes.button} color="primary" aria-label="add" onClick={onClick}>
          <Icon className={classes.icon} children="add" />
        </Fab>
      </Tooltip>
    </div>
  </Card>;

NewStackButton.propTypes = {
  classes: PropTypes.object.isRequired,
  onClick: PropTypes.func.isRequired,
  typeName: PropTypes.string.isRequired,
};

export default withStyles(styles)(NewStackButton);
