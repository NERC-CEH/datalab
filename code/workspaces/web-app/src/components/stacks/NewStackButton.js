import React from 'react';
import PropTypes from 'prop-types';
import PagePrimaryAction from '../common/buttons/PagePrimaryActionButton';

const NewStackButton = ({ onClick, typeName, labelPrefix = 'Create' }) => (
  <PagePrimaryAction aria-label="add" onClick={onClick}>
    {`${labelPrefix} ${typeName}`}
  </PagePrimaryAction>
);

NewStackButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  typeName: PropTypes.string.isRequired,
};

export default NewStackButton;
