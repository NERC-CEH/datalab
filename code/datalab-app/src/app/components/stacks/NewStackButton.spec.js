import React from 'react';
import { createShallow } from 'material-ui/test-utils';
import NewStackButton from './NewStackButton';

function shallowRender(props) {
  const shallow = createShallow({ dive: true });

  return shallow(<NewStackButton {...props} />);
}

describe('New Stack Button', () => {
  const generateProps = () => ({
    onClick: () => {},
    typeName: 'Stack',
  });

  it('creates correct snapshot', () => {
    // Arrange
    const props = generateProps();

    // Act
    const output = shallowRender(props);

    // Assert
    expect(output).toMatchSnapshot();
  });
});
