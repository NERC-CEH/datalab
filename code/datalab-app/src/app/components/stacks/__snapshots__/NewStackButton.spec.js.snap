// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Stack Button creates correct snapshot 1`] = `
<WithStyles(Card)
  className="NewStackButton-card-1"
  elevation={0}
>
  <div
    className="NewStackButton-buttonContainer-2"
  >
    <WithStyles(Tooltip)
      title="Create Stack"
    >
      <WithStyles(Fab)
        aria-label="add"
        className="NewStackButton-button-3"
        color="primary"
        onClick={[Function]}
      >
        <WithStyles(Icon)
          className="NewStackButton-icon-4"
        >
          add
        </WithStyles(Icon)>
      </WithStyles(Fab)>
    </WithStyles(Tooltip)>
  </div>
</WithStyles(Card)>
`;
